src/Eloquent/Model.php

```diff
@@ -294,20 +295,11 @@ public function originalIsEquivalent($key)
             return false;
         }
 
-        if ($this->isDateAttribute($key)) {
-            $attribute = $attribute instanceof UTCDateTime ? $this->asDateTime($attribute) : $attribute;
-            $original = $original instanceof UTCDateTime ? $this->asDateTime($original) : $original;
-
-            return $attribute == $original;
-        }
-
-        if ($this->hasCast($key, static::$primitiveCastTypes)) {
-            return $this->castAttribute($key, $attribute) ===
-                $this->castAttribute($key, $original);
+        if (is_scalar($attribute) || is_scalar($original)) {
+            return false;
         }
 
-        return is_numeric($attribute) && is_numeric($original)
-            && strcmp((string) $attribute, (string) $original) === 0;
+        return fromPHP([$attribute]) === fromPHP([$original]);
```

Comment:
A couple of notes on this:
* `fromPHP` will be deprecated and is replaced by `Document::fromPHP`
* Root-Level encoding is only supported for documents (i.e. any object or array with string keys)
* If `$attribute` happens to be an object (e.g. an embedded document), this conversion will not yield the correct results as only public properties are encoded, which is most likely wrong for any Model
* This may not be an issue, but I'll note that using this comparison `['foo' => 'bar', 'bar' => 'foo']` and `['bar' => 'foo', 'foo' => 'bar']` will not be treated as equal. That said, we don't have a good way to check BSON equality, and one may argue that two BSON documents with the same key/value pairs in different order shouldn't be considered equal.

For the time being, using this comparison may be feasible as long as we can ensure that it doesn't have side effects for embedded documents.